db/lastfm_session.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+import os
+import binascii
+import db
+from sqlalchemy import text
+from lastfm_user import User
+
+
+class Session(object):
+    """ Session class required by the api-compat """
+
+    def __init__(self, row):
+        self.id, userid, self.sid, self.api_key, self.timestamp = row
+        self.user = User.load_by_id(userid)
+
+    @staticmethod
+    def load(session, api_key=None):
+        """ Load the session details from the database.
+            If the api_key is also supplied then verify it as well.
+        """
+        params = {'sid': session}
+        query = "SELECT * FROM session WHERE sid=:sid"
+        if api_key:
+            params['api_key'] = api_key
+            query = "SELECT * FROM session WHERE sid=:sid AND api_key=:api_key"
+
+        with db.engine.connect() as connection:
+            result = connection.execute(text(query), params)
```

Comment:
just as an FYI, we've been skipping these checks in AcousticBrainz because of the amount of boilerplate code it added every time we made a query. I agree that it's a good idea to try and catch these errors, perhaps we can take a look at this again and see if we can find a nice way of doing it